config/abilities/provider_any.rb

```diff
@@ -41,10 +41,13 @@
     end
 
     if user.has_permission?(:monitoring)
-      can [:show], AlertRelatedEvent
+      can [:show], AlertRelatedEvent do |event|
+        user.has_access_to_service?(event.try(:service))
+      end
     end
+
     can [:show], Reports::CsvDataExportEvent do |event|
-      user.admin? && event.recipient.try!(:id) == user.id
+      user.admin? && event.recipient&.id == user.id
```

Comment:
They're actually equivalent. `try!` (mind the bang) will raise a `NoMethodError` in case `recipient` doesn't respond to `id`, as the safe navigation operator does.
You may try it out with something like:
```ruby
123.try!(:upcase) #=> NoMethodError
123&.upcase #=> NoMethodError
nil.try!(:update) #=> nil
nil&.update #=> nil
```